### PR TITLE
Apply all changes in master to EE4J_8

### PR DESCRIPTION
**Pull request by markt-asf**

### CODE REVIEW

File: api/client/src/main/java/javax/websocket/Endpoint.java [hunk 1]
```diff
@@ -16,44 +17,41 @@
...
- * <p>If the developer provides a custom {@link javax.websocket.server.ServerEndpointConfig.Configurator}
- * which overrides the default policy for endpoint instance creation, for example, 
- * using a single Endpoint instance for multiple client connections, the developer
- * may need to write code that can execute concurrently.
+ * <p>
+ * Here is an example of a simple endpoint that echoes any incoming text message back to the sender.
  *
- * <p>Here is an example of a simple endpoint that echoes any incoming text message back to the sender.
- * <pre><code>
+ * <pre>
```
Comment (joakime) [inline api/client/src/main/java/javax/websocket/Endpoint.java]: This `<p>` was skipped.

File: api/server/src/main/java/javax/websocket/server/ServerEndpointConfig.java [hunk 2]
```diff
@@ -109,136 +105,124 @@
...
+     * <p>
+     * Here are some examples:
      *
-     * <p>Here are some examples:
+     * <p>
+     * Building a plain configuration for an endpoint with just a path.
      *
-     * <p>Building a plain configuration for an endpoint with just a path.
-     * <pre><code>
+     * <pre>
```
Comment (joakime) [inline api/server/src/main/java/javax/websocket/server/ServerEndpointConfig.java]: Another `<p>` was dropped in this merge.

File: api/server/src/main/java/javax/websocket/server/ServerEndpointConfig.java [hunk 3]
```diff
@@ -268,25 +252,18 @@ public static Builder create(Class<?> endpointClass, String path) {
...
-                    this.endpointClass,
-                    this.path,
-                    Collections.unmodifiableList(this.subprotocols),
-                    Collections.unmodifiableList(this.extensions),
-                    Collections.unmodifiableList(this.encoders),
-                    Collections.unmodifiableList(this.decoders),
-                    this.serverEndpointConfigurator
-                 );
+            return new DefaultServerEndpointConfig(this.endpointClass, this.path, this.subprotocols, this.extensions,
+                    this.encoders, this.decoders, this.serverEndpointConfigurator);
```
Comment (joakime) [inline api/server/src/main/java/javax/websocket/server/ServerEndpointConfig.java]: This looks like an API change.
Missing the `Collections.unmodifiableList` changes from master.

**Comment by markt-asf** [comment]
I'll provide PRs to restore those lost comments shortly.
The removal of `Collections.unmodifiableList()` was deliberate. See #179. I don't believe it introduces an API change.

**Comment by markt-asf** [comment]
@joakime  - are you are to merge this? I'm seeing a message suggesting I am not an authorised user.

**Comment by markt-asf** [comment]
I've double checked and those Javadoc comments aren't lost, just reformatted. There are inserts for the same comment a few lines above the deletion in the diff.

**Comment by joakime** [comment]
websocket-api-restricted-branch

@markt-asf it seems I'm not authorized as well.
@m0mus do you have the rights to modify the `EE4J-8` branch restrictions on this repo?

**Comment by senivam** [comment]
the name of the whole branch is wrong. It shall be called EE4J_8

**Comment by joakime** [comment]
@senivam the branch name `EE4J-8` was created by @m0mus on Jan 18, 2018.
See https://github.com/eclipse-ee4j/websocket-api/branches

Is the name the reason for the restrictions on merge for @markt-asf and myself?

**Comment by joakime** [comment]
Still reports "You're not authorized to merge this pull request" for me.
What about you @markt-asf ? do you have rights now?

**Comment by joakime** [comment]
@m0mus thanks for the fix to the permissions.

**Comment by joakime** [comment]
@senivam the EE4J-8 pom has the SNAPSHOT entries.

https://github.com/eclipse-ee4j/websocket-api/blob/EE4J-8/api/pom.xml#L37

https://github.com/eclipse-ee4j/websocket-api/blob/545375e5cbda1dc6454dd571533237d23232c7d8/api/pom.xml#L37

**Comment by senivam** [comment]
aha, it's already fine, perfect. so, all is done, thank you :)

**Comment by senivam** [comment]
dry run of release job at https://jenkins.eclipse.org/websocket/job/websocket-api-release-build/ for EE4J-8 branch passed OK, so we can close the issue #259

**Comment by joakime** [comment]
@senivam if you want to kick off the release build, feel free.
Otherwise let me know, I'll try to take care of it.

**Pull request merged**